node/pkg/accounting/accounting.go

```diff
@@ -0,0 +1,373 @@
+)
+
+const (
+	MainNetMode = 1
+	TestNetMode = 2
+	DevNetMode  = 3
+	GoTestMode  = 4
+
+	// We will retry requests once per minute for up to an hour.
+	auditInterval = time.Duration(time.Minute)
+	maxRetries    = 60
+)
+
+type (
+	// tokenBridgeKey is the key to the map of token bridges being monitored
+	tokenBridgeKey struct {
+		emitterChainId vaa.ChainID
+		emitterAddr    vaa.Address
+	}
+
+	// tokenBridgeEntry is the payload of the map of the token bridges being monitored
+	tokenBridgeEntry struct {
+	}
+
+	// pendingEntry is the payload for each pending transfer
+	pendingEntry struct {
+		msg        *common.MessagePublication
+		msgId      string
+		digest     string
+		updTime    time.Time
+		retryCount int
+	}
+)
+
+// Accounting is the object that manages the interface to the wormchain accounting smart contract.
+type Accounting struct {
+	ctx              context.Context
+	logger           *zap.Logger
+	db               db.AccountingDB
+	contract         string
+	wsUrl            string
+	lcdUrl           string
+	wormchainConn    *wormconn.ClientConn
+	enforceFlag      bool
+	gk               *ecdsa.PrivateKey
+	gst              *common.GuardianSetState
+	msgChan          chan<- *common.MessagePublication
+	mutex            sync.Mutex
+	tokenBridges     map[tokenBridgeKey]*tokenBridgeEntry
+	pendingTransfers map[string]*pendingEntry // Key is the message ID (emitterChain/emitterAddr/seqNo)
+	env              int
+}
+
+// NewAccounting creates a new instance of the Accounting object.
+func NewAccounting(
+	ctx context.Context,
+	logger *zap.Logger,
+	db db.AccountingDB,
+	contract string, // the address of the smart contract on wormchain
+	wsUrl string, // the URL of the wormchain websocket interface
+	lcdUrl string, // the URL of the wormchain LCD interface
+	wormchainConn *wormconn.ClientConn, // used for communicating with the smart contract
+	enforceFlag bool, // whether or not accounting should be enforced
+	gk *ecdsa.PrivateKey, // the guardian key used for signing observation requests
+	gst *common.GuardianSetState, // used to get the current guardian set index when sending observation requests
+	msgChan chan<- *common.MessagePublication, // the channel where transfers received by the accounting runnable should be published
+	env int, // Controls the set of token bridges to be monitored
+) *Accounting {
+	return &Accounting{
+		ctx:              ctx,
+		logger:           logger,
+		db:               db,
+		contract:         contract,
+		wsUrl:            wsUrl,
+		lcdUrl:           lcdUrl,
+		wormchainConn:    wormchainConn,
+		enforceFlag:      enforceFlag,
+		gk:               gk,
+		gst:              gst,
+		msgChan:          msgChan,
+		tokenBridges:     make(map[tokenBridgeKey]*tokenBridgeEntry),
+		pendingTransfers: make(map[string]*pendingEntry),
+		env:              env,
+	}
+}
+
+// Run initializes the accounting module and starts the watcher runnable.
+func (acct *Accounting) Run(ctx context.Context) error {
```

Comment:
naming nit: This is called `Run` but it really just starts the watcher and the `Run` function is already overloaded for supervisor runnables.  How about we call this `Start`?